### PR TITLE
[LOI editor] Sort LOIs alphabetically in list

### LINKED ISSUE #1672: Imported points don't show in alphabetical order
**Describe the bug**
Imported points don't show in alphabetical order after upload

image

### PR DESCRIPTION
**Pull request by hallsjj08**
Fixes #1672

Added an orderBy function provided by the FireStore library to the collection reference when retrieving LOI's using the DataStoreService.

[LOI_Recording.webm](https://github.com/google/ground-platform/assets/4734964/fa9a01e1-376b-4f2e-a85a-e5e34e5e732e)

### CODE REVIEW

File: web/src/app/services/data-store/data-store.service.ts [hunk 1]
```diff
@@ -298,7 +298,7 @@ export class DataStoreService {
 
   lois$({id}: Survey): Observable<List<LocationOfInterest>> {
     return this.db
-      .collection(`${SURVEYS_COLLECTION_NAME}/${id}/lois`)
+      .collection(`${SURVEYS_COLLECTION_NAME}/${id}/lois`, ref => ref.orderBy('properties.name'))
```
Comment (gino-m) [inline web/src/app/services/data-store/data-store.service.ts:301]: Thanks for the fix! The label shown in the UIs is a composite of name and ID, so you may need to sort this somewhere further down the stack. `loi.service.ts` has `getDisplayName()` which can be used to form the name for sorting. You can sort them in the lois exposed from that service, or further down in the UI layer if you think it's appropriate.

**Review by gino-m — CHANGES_REQUESTED** [review]

**Comment by gino-m** [comment]
@hallsjj08 Gentile ping, please let me know you have any questions!

File: web/src/app/services/data-store/data-store.service.ts (hunk 1 above)
Comment (hallsjj08) [reply web/src/app/services/data-store/data-store.service.ts]: I moved the sorting into the `loi.service.ts` file in the `getLocationsOfInterest$ `method. Sorry it took me so long to get around to it. I had a lot going on for a bit there, but I should be free to help out more often. So feel free to assign me some more issues. Also, this is my first time using RxJS, so let me know if you think there's a better place to put this than where it's at.

**Review by gino-m — DISMISSED** [review]
Thanks for the contribution! LGTM, assuming checks pass!

**Comment by gino-m** [comment]
@hallsjj08 Looks like format lint failed - please run `npm run lint:fix` or use Prettier in your IDE to fix!

**Review by gino-m — APPROVED** [review]
LGTM!

**Review by gino-m — CHANGES_REQUESTED** [review]
@hallsjj08 Thanks for the updates. Unfortunately it seems the formatting still doesn't match what our linter expects. Did you use and IDE plugin by chance? Maybe `npm run lint:fix` will give a better result. Here's what CI is reporting:

```
Linting "web"...

/home/runner/work/ground-platform/ground-platform/web/src/app/converters/firebase-data-converter.ts
Warning:   254:44  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
Warning:   522:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
Lint warnings found in the listed files.

Lint errors found in the listed files.


/home/runner/work/ground-platform/ground-platform/web/src/environments/environment.dev.ts
Error:   17:1  error  There should be no empty line within import group  import/order

/home/runner/work/ground-platform/ground-platform/web/src/environments/environment.prod.ts
Error:   17:1  error  There should be no empty line within import group  import/order

✖ 4 problems (2 errors, 2 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.

Error: Process completed with exit code 1.
```

**Comment by hallsjj08** [comment]
> @hallsjj08 Thanks for the updates. Unfortunately it seems the formatting still doesn't match what our linter expects. Did you use and IDE plugin by chance? Maybe `npm run lint:fix` will give a better result. Here's what CI is reporting:
> 
> ```
> Linting "web"...
> 
> /home/runner/work/ground-platform/ground-platform/web/src/app/converters/firebase-data-converter.ts
> Warning:   254:44  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
> Warning:   522:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
> Lint warnings found in the listed files.
> 
> Lint errors found in the listed files.
> 
> 
> /home/runner/work/ground-platform/ground-platform/web/src/environments/environment.dev.ts
> Error:   17:1  error  There should be no empty line within import group  import/order
> 
> /home/runner/work/ground-platform/ground-platform/web/src/environments/environment.prod.ts
> Error:   17:1  error  There should be no empty line within import group  import/order
> 
> ✖ 4 problems (2 errors, 2 warnings)
>   2 errors and 0 warnings potentially fixable with the `--fix` option.
> 
> Error: Process completed with exit code 1.
> ```

@gino-m, it looks like the linter didn't recognize the import of `firebaseConfig` in the environment files so the `lint:fix` added a new line before the next import. I created a stub config to resolve the error in my local environment so that it doesn't happen again.  I'm still getting the lint warning for using `any` type tho. I haven't touched these files however. Does the check also need to resolve these warnings or only the errors?

**Review by gino-m — APPROVED** [review]

**Comment by hallsjj08** [comment]
@gino-m, it looks like I'm unauthorized to merge this PR.

**Comment by gino-m** [comment]
Merged! Nice work!

**Pull request merged**